dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1122,12 +1123,30 @@ void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const
 
             if (f6Button)
             {
-                auto const scopedRevoker = f6Button.GettingFocus(winrt::auto_revoke, [this](auto const&, auto const& args) {
+                auto const scopedRevoker = f6Button.GettingFocus(winrt::auto_revoke, [this](auto const&, const winrt::GettingFocusEventArgs& args) {
                     m_previouslyFocusedElement = winrt::make_weak(args.OldFocusedElement());
                 });
                 const bool setFocus = f6Button.Focus(winrt::FocusState::Keyboard);
                 args.Handled(setFocus);
             }
+            // If there is no close button, try setting focus on the tip's custom content.
+            else if (const auto mainContentPresenter = m_mainContentPresenter.get())
```

Comment:
Instead of making a seperate else block for this case, can we just expand the lambda that sets the f6Button a bit? something like:
```
            const winrt::Button f6Button = [this]() -> winrt::Button
            {
                auto firstButton = m_closeButton.get();
                auto secondButton = m_alternateCloseButton.get();
                //Prefer the close button to the alternate, except when there is no content.
                if (!CloseButtonContent())
                {
                    std::swap(firstButton, secondButton);
                }
                if (firstButton && firstButton.Visibility() == winrt::Visibility::Visible)
                {
                    return firstButton;
                }
                else if (secondButton && secondButton.Visibility() == winrt::Visibility::Visible)
                {
                    return secondButton;
                }
else if (const auto mainContentPresenter = m_mainContentPresenter.get())
{
    return const auto focusableElement = winrt::FocusManager::FindFirstFocusableElement(mainContentPresenter);
}
                return nullptr;
            }();
```
Then we shouldn't need this block at all

Reply:
@StephenLPeters Moved code into lambda as suggested. Note that I currently am solely using the FocusManager::GetFocusedElement() API to get the previously focused element as as the `GettingFocus` event is not available to all possibly focusably elements (such as a Hyperlink). 

That said, if you prefer we use the [UIElement.GettingFocus](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.gettingfocus?view=winrt-19041) API wherever possible instead (and the documentation [suggests](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.focusmanager?view=winrt-19041#remarks) this), I can that use case as well (i.e. whenever our *f6FocusableElement* is a `UIElement` we use the GettingFocus event instead).